### PR TITLE
dev-8412 Nested Dropdown Enhancement - Dashboard Level

### PR DESCRIPTION
**Pull request by Matt-DiPaolo**
## Self Review

- I have added testing steps for reviewers
- I have commented my code, particularly in hard-to-understand areas
- My changes generate no new warnings
- New and existing unit tests are passing

### CODE REVIEW

File: packages/chart/src/_stories/Chart.stories.tsx [hunk 1]
```diff
@@ -22,7 +22,7 @@ export const line_Chart_Two_Points_Regression_Test: Story = {
 export const line_Chart_Two_Points_New_Chart: Story = {
   args: {
     config: lineChartTwoPointsNewChart,
-    isEditor: false
+    isEditor: true
```
Comment (joshlacey) [inline packages/chart/src/_stories/Chart.stories.tsx:25]: pull this file out of your commit history for this pr

File: packages/chart/src/components/EditorPanel/EditorPanel.tsx [hunk 2]
```diff
@@ -1444,19 +1444,21 @@ const EditorPanel = () => {
...
-                    )}
-                    {config.dynamicSeries && config.visualizationType === 'Line' && config.dynamicSeriesType === 'Line' && (
-                      <Select
-                        fieldName='dynamicSeriesLineType'
-                        value={config.dynamicSeriesLineType ? config.dynamicSeriesLineType : 'curveLinear'}
-                        label='Line Type'
-                        initial='Select'
-                        updateField={updateField}
-                        options={Object.keys(approvedCurveTypes).map(curveName => approvedCurveTypes[curveName])}
+                        options={['Line', 'dashed-sm', 'dashed-md', 'dashed-lg']}
```
Comment (joshlacey) [inline packages/chart/src/components/EditorPanel/EditorPanel.tsx:1447]: I'm not seeing any changes in this file, if none can you remove from your commit history?

File: packages/chart/src/components/LineChart/index.tsx [hunk 3]
```diff
@@ -60,11 +60,25 @@ const LineChart = (props: LineChartProps) => {
           let lineType = config.runtime.series.filter(item => item.dataKey === seriesKey)[0].type
           const seriesData = config.runtime.series.filter(item => item.dataKey === seriesKey)
           const seriesAxis = seriesData[0].axis ? seriesData[0].axis : 'left'
-          let displayArea = legend.behavior === 'highlight' || seriesHighlight.length === 0 || seriesHighlight.indexOf(seriesKey) !== -1
+          let displayArea =
```
Comment (joshlacey) [inline packages/chart/src/components/LineChart/index.tsx:63]: I'm not seeing any changes in this file, if none can you remove from your commit history?

**Review by joshlacey — CHANGES_REQUESTED** [review]
@Matt-DiPaolo can you take a second look at your commits.

We added a new rule to the linter which is autoformatting things on save. Make sure that if you save a file that it's saved because you need the change for this feature. otherwise please pull those changes out of your commit history. I highlighted three files that seemed to have that issue.

File: packages/core/components/Filters.tsx [hunk 4]
```diff
@@ -277,7 +277,7 @@ type FilterProps = {
 
 const Filters = (props: FilterProps) => {
   const { config: visualizationConfig, filteredData, dimensions } = props
-  const { filters, type, general, theme, filterBehavior } = visualizationConfig
+  const { data, filters, type, general, theme, filterBehavior } = visualizationConfig
```
Comment (joshlacey) [inline packages/core/components/Filters.tsx:280]: why is data being added here?

File: packages/core/components/Filters.tsx [hunk 5]
```diff
@@ -378,8 +378,12 @@ const Filters = (props: FilterProps) => {
       const { active, queuedActive, label, filterStyle } = singleFilter as VizFilter
 
       handleSorting(singleFilter)
+      singleFilter.orderedValues =
+        singleFilter.orderedValues === undefined || singleFilter.orderedValues.length === 0
+          ? singleFilter.values
+          : singleFilter.orderedValues
 
-      singleFilter.values?.forEach((filterOption, index) => {
+      singleFilter.orderedValues?.forEach((filterOption, index) => {
```
Comment (joshlacey) [inline packages/core/components/Filters.tsx:386]: I don't think this is necessary. values should be accurate. if there are orderedValues then handleSorting should have fixed that.

File: packages/dashboard/src/_stories/Dashboard.stories.tsx [hunk 6]
```diff
@@ -30,6 +31,12 @@ const meta: Meta<typeof Dashboard> = {
 
 type Story = StoryObj<typeof Dashboard>
 
+export const Dashboard_Filter_Build: Story = {
```
Comment (joshlacey) [inline packages/dashboard/src/_stories/Dashboard.stories.tsx:34]: Can you give this a name more specific to Nested Dropdown?

File: packages/dashboard/src/_stories/Dashboard.stories.tsx [hunk 7]
```diff
@@ -61,7 +68,7 @@ export const Bump_Chart_Dashboard: Story = {
 export const Dashboard_Filters: Story = {
   args: {
     config: Dashboard_Filter,
-    isEditor: false
+    isEditor: true
```
Comment (joshlacey) [inline packages/dashboard/src/_stories/Dashboard.stories.tsx:71]: false

File: packages/dashboard/src/_stories/_mock/dashboard-filter-build.json [hunk 8]
```diff
@@ -0,0 +1,2839 @@
+{
```
Comment (joshlacey) [inline packages/dashboard/src/_stories/_mock/dashboard-filter-build.json:1]: @Matt-DiPaolo actually can we delete out all of the data that isn't being used? should be able to do it with a regex.

File: packages/dashboard/src/components/DashboardFilters/DashboardFilters.tsx [hunk 9]
```diff
@@ -64,7 +69,7 @@ const DashboardFilters: React.FC<DashboardFilterProps> = ({
           })
         }
 
-        return filter.multiSelect ? (
+        return filter.filterStyle === 'multiselect' ? (
```
Comment (joshlacey) [inline packages/dashboard/src/components/DashboardFilters/DashboardFilters.tsx:72]: this may cause issues with existing configs that don't have filterStyle set.

File: packages/dashboard/src/scss/main.scss [hunk 10]
```diff
@@ -262,6 +262,10 @@
     width: 100%;
   }
 
+  .cove-dashboard-filters-container {
+    z-index: 5;
```
Comment (joshlacey) [inline packages/dashboard/src/scss/main.scss:266]: why is this necessary?

File: packages/dashboard/src/components/DashboardFilters/DashboardFilters.tsx (hunk 9 above)
Comment (joshlacey) [reply packages/dashboard/src/components/DashboardFilters/DashboardFilters.tsx:72]: to be safe let's add a migration script for this.

**Review by joshlacey — CHANGES_REQUESTED** [review]

File: packages/dashboard/src/scss/main.scss (hunk 10 above)
Comment (Matt-DiPaolo) [reply packages/dashboard/src/scss/main.scss:266]: Without this, the graph has a higher z-index and will cover the dropdown when it is opened. Since this isn't an html Select element, the z-index is set to be like a div and the proceeding div covers it up.

File: packages/core/components/EditorPanel/VizFilterEditor/VizFilterEditor.tsx [hunk 11]
```diff
@@ -80,7 +81,7 @@ const VizFilterEditor: React.FC<VizFilterProps> = ({ config, updateField, rawDat
     const filterItem = { ...filtersCopy[filterIndex] }
 
     // Overwrite filterItem.values since thats what we map through in the editor panel
-    filterItem.values = updatedValues
+    filterItem.active = updatedValues[0]
```
Comment (joshlacey) [inline packages/core/components/EditorPanel/VizFilterEditor/VizFilterEditor.tsx:84]: why did you make this change? it's already being set on line 86??

File: packages/core/components/Filters.tsx [hunk 12]
```diff
@@ -378,8 +378,12 @@ const Filters = (props: FilterProps) => {
       const { active, queuedActive, label, filterStyle } = singleFilter as VizFilter
 
       handleSorting(singleFilter)
+      singleFilter.orderedValues =
```
Comment (joshlacey) [inline packages/core/components/Filters.tsx:381]: Why did you make this change? did you notice a bug in the viz filters?

File: packages/dashboard/src/helpers/filterData.ts [hunk 13]
```diff
@@ -35,7 +35,12 @@ function filter(data = [], filters: SharedFilter[], condition) {
         isNotTheSelectedValue = selectedValue && currentValue != selectedValue
       }
 
-      if ((filter.filterStyle === 'nested-dropdown' && filter.subGrouping && filter.active === currentValue, isNotTheSelectedValue === false)) {
+      if (
```
Comment (joshlacey) [inline packages/dashboard/src/helpers/filterData.ts:38]: @Matt-DiPaolo can you make sure this is right? it was resolving to always true otherwise.

Comment (Matt-DiPaolo) [reply packages/dashboard/src/helpers/filterData.ts:38]: I may be misunderstanding, but it is not always resolving to true. It is only true when:
- selectedValue is not an array
- selectedValue is undefined
- currentValue doesn't equal selectedValue

Since nested dropdowns is working with a subgroup. The currentValue and the selected Value can equal each other. e.g. **Age Group - 56 years old** and **Age Group - Overall** have the same currentValue and selectedValue of "Age Group." This example would set isNotTheSelectedValue to false.

If it is a nested dropdown and the isNotTheSelectedValue is false, then set isNotTheSelectedValue to true if:
- subGroupActive is undefined 
- subGroupActive doesn't equal the selectedSubGroupValue

I am going to remove the 
 `
if (subGroupActive === undefined || value !== subGroupActive) {
          isNotTheSelectedValue = true
        } 
`
and replace with 
` isNotTheSelectedValue = subGroupActive && selectedSubGroupValue !== subGroupActive`

File: packages/core/components/Filters.tsx [hunk 14]
```diff
@@ -378,8 +378,7 @@ const Filters = (props: FilterProps) => {
       const { active, queuedActive, label, filterStyle } = singleFilter as VizFilter
 
       handleSorting(singleFilter)
-
-      singleFilter.values?.forEach((filterOption, index) => {
+      singleFilter.values.forEach((filterOption, index) => {
```
Comment (joshlacey) [inline packages/core/components/Filters.tsx:381]: can you remove this file from the Commit history? @Matt-DiPaolo

File: packages/core/components/EditorPanel/VizFilterEditor/VizFilterEditor.tsx [hunk 15]
```diff
@@ -80,7 +81,6 @@ const VizFilterEditor: React.FC<VizFilterProps> = ({ config, updateField, rawDat
     const filterItem = { ...filtersCopy[filterIndex] }
 
     // Overwrite filterItem.values since thats what we map through in the editor panel
-    filterItem.values = updatedValues
```
Comment (joshlacey) [inline packages/core/components/EditorPanel/VizFilterEditor/VizFilterEditor.tsx:83]: @Matt-DiPaolo why as values removed in the first place? the comment above it should be updated if it needs to be removed. 

just not sure why we are making any changes to the VizFilterEditor unless there was an actual bug you found

File: packages/chart/src/CdcChart.tsx [hunk 16]
```diff
@@ -88,7 +89,10 @@ export default function CdcChart({
   const [loading, setLoading] = useState(true)
   const [colorScale, setColorScale] = useState(null)
   const [config, setConfig] = useState<ChartConfig>({} as ChartConfig)
-  const [stateData, setStateData] = useState(config.data || [])
+  const [stateData, _setStateData] = useState(configObj.data || [])
```
Comment (joshlacey) [inline packages/chart/src/CdcChart.tsx:92]: @Matt-DiPaolo I think you overwrote my commit that reverted this.

File: packages/chart/src/components/LinearChart.tsx [hunk 17]
```diff
@@ -74,7 +74,13 @@ const LinearChart: React.FC<LinearChartProps> = props => {
```
[the remark this answers is not in the document]
Comment (joshlacey) [reply packages/chart/src/components/LinearChart.tsx:74]: sorry this was my change. I forgot :)

File: packages/core/components/EditorPanel/VizFilterEditor/VizFilterEditor.tsx [hunk 18]
```diff
@@ -52,7 +52,8 @@ const VizFilterEditor: React.FC<VizFilterProps> = ({ config, updateField, rawDat
 
   const handleNameChange = (filterIndex, columnName) => {
     const values = _.uniq(rawData.map(row => row[columnName]))
-    const copiedFilter = { ..._.cloneDeep(config.filters[filterIndex]), columnName, values }
+    const orderedValues = values
```
Comment (joshlacey) [inline packages/core/components/EditorPanel/VizFilterEditor/VizFilterEditor.tsx:55]: @Matt-DiPaolo is this a fix?

File: packages/core/components/Filters.tsx (hunk 12 above)
Comment (joshlacey) [reply packages/core/components/Filters.tsx:381]: It looks like you removed this and then re-added it?

**Review by joshlacey — APPROVED** [review]
